src/phoenix/datasets/experiments.py

```diff
@@ -329,13 +332,24 @@ def evaluate_experiment(
 
 def _evaluate_experiment(
     experiment: Experiment,
-    evaluators: Union[ExperimentEvaluator, Iterable[ExperimentEvaluator]],
+    evaluators: Union[EvaluatorOrCallable, Iterable[EvaluatorOrCallable]],
```

Comment:
I'm confused on this input type. This type suggests we should accept raw (undecorated functions), but shouldn't the type after decorating a function with the `create_factory` be `Callable[[], ExperimentEvaluator]`?

Reply:
we will auto-decorate bare functions passed into run_experiment (and evaluate_experiment)

Reply:
For example, this script:

```python
from phoenix.datasets.evaluators.utils import create_evaluator
from phoenix.datasets.types import CanEvaluate


@create_evaluator()
def factory(x: int) -> bool:
    return x < 3


f = factory
print(f"{type(f)=}")
print(f"{isinstance(f, CanEvaluate)=}")
```

results in this output:

```
type(f)=<class 'phoenix.datasets.evaluators.utils._wrap_sync_evaluation_function.<locals>.wrapper.<locals>.SyncEvaluator'>
isinstance(f, CanEvaluate)=True
```

I think this is getting masked by the `cast`.

Reply:
Oh, it looks like we intend to let people pass in raw functions.